Project 4/README.md

```diff
@@ -45,3 +45,133 @@ You're probably thinking, "why would I submit a level 8 kata if they're not wort
+}
+Kata level 5 – 11 points
+Write a SPARQL query that CONSTRUCTS a new graph of all books that have been written by authors who have also written books in the “Science Fiction” genre. The new graph should include the title, author, and genre of each book, as well as the name and birth year if each author.
+PREFIX rdf: http://www.w3.org/1999/02/22-rdf-syntax-ns#
+PREFIX dc: http://purl.org/dc/terms/
+PREFIX foaf: http://xmlns.com/foaf/0.1/
+PREFIX genre http://example.com/genre#
+
+CONSTRUCT {
+ ?book rdf:type dc:Book ;
+dc:title >?title ;
+dc:creator ?author ;
+genre:ScienceFiction ?genre . 
+?author rdf: type foaf:Person ;
+	foaf:name ?name ;
+	foaf:birthyear ?birthYear . 
+}
+WHERE{
+?book rdf:type dc:Book ;
+	dc:title ?title
+	dc:creator ?author ;
+	genre:Sciencefiction ?genre . 
+?author rdf:type foaf:Person ;
+	foaf:name ?name ; 
+	foaf:birthyear ?birthYear . 
+FILTER EXISTS {
+?otherBook rdf:type dc:Book ;
+	dc:creator ?author ; 
+genre:ScienceFicton ?genre . 
+Filter (?otherBook !=?book)
+}
+}
+
+Kata level 5? – 16 points 
+Using dbpedia write a SPARQL query that CONSTRUCTS a new graph of all horror movies that have been released after January 1st, 2015 along with their titles, directors, and ratings. The WHERE clause is used to select the horror movies that meet the criteria, and the FILTER clause is used to filter the results by release date.
+CONSTRUCT { 
+?movie a 
+<http://schema.org/Movie> ; 
+<http://schema.org/name> ?title ; 
+<http://schema.org/director> ?director ; 
+<http://schema.org/contentRating> ?rating .
+ } 
+WHERE { 
+?movie a <http://schema.org/Movie> ;
+<http://schema.org/name> ?title ; 
+<http://schema.org/director> ?director ; 
+http://schema.org/genre
+ <http://schema.org/Horror> ; 
+<http://schema.org/contentRating> ?rating ; 
+<http://schema.org/datePublished> ?date . 
+FILTER(?date > "2015-01-01"
+^^<http://www.w3.org/2001/XMLSchema#date>)
+ }
+
+Kata level 6? -- 19 points
+This query searches for 10 books in DBpedia that are considered cosmic horror and were published after January 1st, 1900. It returns the author, book title, book description, publisher, genre, and publication date for each book
+PREFIX dbpedia-owl: <http://dbpedia.org/ontology/>
+PREFIX dcterms: <http://purl.org/dc/terms/>
+PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+PREFIX dbo: <http://dbpedia.org/ontology/>
+PREFIX dbp: <http://dbpedia.org/property/>
+
+SELECT ?author ?book ?description ?publisher ?genre ?publicationDate
+WHERE {
+  ?book a dbo:Book ;
+        dcterms:description ?description ;
+        dbp:genre "Cosmic horror"@en ;
+        dbo:author ?author ;
+        dbo:publisher ?publisher ;
+        dbo:literaryGenre ?genre ;
+        dbo:publicationDate ?publicationDate .
+  FILTER regex(?description, "cosmic horror", "i")
+  FILTER regex(?genre, "horror", "i")
+  FILTER (?publicationDate >= "1900-01-01"^^xsd:date)
+}
+ORDER BY DESC(?publicationDate)
+LIMIT 10
+
+Level 5? -- 21 points
```

Comment:
This is ok but I would say it is a level 6, not a level 5. For a level 5 I am thinking of a bit more complex queries with multiple "tricks" (filter, regex, comparison of values, etc.)

Reply:
hmmm thank you. Maybe I can further complicate it...